src/init.cpp

```diff
@@ -1031,6 +1031,19 @@ bool AppInitParameterInteraction(const ArgsManager& args, bool use_syscall_sandb
         return InitError(_("No proxy server specified. Use -proxy=<ip> or -proxy=<ip:port>."));
     }
 
+    if (args.IsArgSet("-reindex-chainstate")) {
```

Comment:
In commit "init: disallow reindex-chainstate with optional indexes" (93fa8f651242af5354ba92f7aaec18e878d4f840)

It would be good to replace `args.IsArgSet("-reindex-chainstate")` with `args.GetBoolArg("-reindex-chainstate", false)` because otherwise this will wrongly trigger for `-noreindex-chainstate`.

(In general, it's good to avoid the `IsArgSet` function. It's frequently misused and leads to minor bugs like this.)

Reply:
Done as suggested, good catch! Maybe there could be a warning about this in the doc of `IsArgSet`  (`system.h`).